dist/prototypes.py

```diff
@@ -99,42 +99,31 @@ def prototypes_extern():
             continue
         if fnmatch.fnmatch(name, '*/checksum/zseries/*'):
             continue
-        if fnmatch.fnmatch(name, '*/os_posix/*'):
-            continue
-        if fnmatch.fnmatch(name, '*/os_win/*'):
+        if re.match(r'^.*/os_(?:posix|win|linux|darwin)/.*', name):
+            # Handled separately in prototypes_os().
             continue
         if fnmatch.fnmatch(name, '*/ext/*'):
             continue
         fn_prototypes(fns, tests, name)
 
     output(fns, tests, "../src/include/extern.h")
 
-# Update POSIX-specific function prototypes.
-def prototypes_posix():
-    fns = []
-    tests = []
-    for name in source_files():
-        if not fnmatch.fnmatch(name, '*.c') + fnmatch.fnmatch(name, '*_inline.h'):
-                continue;
-        if not fnmatch.fnmatch(name, '*/os_posix/*'):
-            continue
-        fn_prototypes(fns, tests, name)
-
-    output(fns, tests, "../src/include/extern_posix.h")
-
-# Update Windows-specific function prototypes.
-def prototypes_win():
-    fns = []
-    tests = []
+def prototypes_os():
+    """
+    The operating system abstraction layer duplicates function names. So each 
+    os gets its own extern header file.
+    """
+    ports = 'posix win linux darwin'.split()
```

Comment:
We're relying on this enumeration in a bunch of places now - perhaps it should be somewhere common? I know Peter has a PR for moving common functionality around, it may already be merged.

Reply:
Yeah look, sorry I don't know where it should live. Feels like it should be a separate ticket.

Reply:
Fair enough - I can see a case that it's part of the build system. Do you want me to make the ticket since I raised the objection?

Reply:
@marcbutler Friendly bump given you've requested a re-review.

Reply:
> Yes: as you're clear on your ask here.

Yes I would like you to make the ticket.

Reply:
Created: [WT-13192](https://jira.mongodb.org/browse/WT-13192)

Reply:
Done, WT-13193. Sorry I missed the earlier request.